File: JPA/projects/ex1-hello-jpa/src/main/java/hellojpa/Member.java
package hellojpa;

import jakarta.persistence.Embedded;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;

@Entity
public class Member {

    @Id
    @GeneratedValue
    private Long id;
    private String name;
    private int age;

    @Embedded
    private Period workPeriod; // 근무 기간

    @Embedded
    private Address homeAddress; // 집 주소

}

File: projects/kotlin/kt-security-example/src/main/kotlin/com/example/ktsecurityexample/member/entity/Member.kt
package com.example.ktsecurityexample.member.entity

import com.example.ktsecurityexample.common.status.Gender
import jakarta.persistence.*
import java.time.LocalDate

@Entity
@Table(
    name = "member",
    uniqueConstraints = [
        UniqueConstraint(name = "uk_member_login_id", columnNames = ["loginId"])
    ]
)
class Member(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "member_id")
    val id: Long? = null,

    @Column(nullable = false, length = 30, updatable = false)
    val loginId: String,

    @Column(nullable = false, length = 100)
    val password: String,

    @Column(nullable = false, length = 10)
    val name: String,

    @Column(nullable = false)
    @Temporal(TemporalType.DATE) // 날짜만 입력
    val birthDate: LocalDate,

    @Column(nullable = false, length = 10)
    @Enumerated(EnumType.STRING)
    val gender: Gender,

    @Column(nullable = false, length = 30)
    val email: String,
)

